File: AppDelivery/app/src/main/java/com/devheverton/appdelivery/Lista_Produtos.java
package com.devheverton.appdelivery;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.google.firebase.auth.FirebaseAuth;

public class Lista_Produtos extends AppCompatActivity {

    private RecyclerView recyclerView_produtos;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_lista_produtos);

recyclerView_produtos = findViewById(R.id.recycleView_produtos);

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_principal, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        int itemID = item.getItemId();

        if (itemID == R.id.perfil){
            Intent intent = new Intent(Lista_Produtos.this,Perfil_Usuario.class);
            startActivity(intent);
            
        } else if (itemID == R.id.pedidos) {
            
        } else if (itemID == R.id.deslogar) {
            FirebaseAuth.getInstance().signOut();
            Toast.makeText(Lista_Produtos.this, "Usuário Deslogado" ,Toast.LENGTH_LONG).show();
            Intent intent = new Intent(Lista_Produtos.this,Form_Login.class);
            startActivity(intent);
            finish();

        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/devhevertonaraujo/mvp/Ui/ListaProdutos/ListaProdutos.kt
package com.devhevertonaraujo.mvp.Ui.ListaProdutos

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.devhevertonaraujo.mvp.Adapter.AdapterProduto
import com.devhevertonaraujo.mvp.Model.AddProdutos
import com.devhevertonaraujo.mvp.Model.Produto
import com.devhevertonaraujo.mvp.R

class ListaProdutos : AppCompatActivity(), ListaProdutosContrato.View {

    private lateinit var presenter: ListaProdutosPresenter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_lista_produtos)

        presenter = ListaProdutosPresenter(this,AddProdutos())
        presenter.MostrarListaProdutos()


        }

    override fun AddProdutos(produtos: MutableList<Produto>) {

        val recycler_produtos = findViewById<RecyclerView>(R.id.recycler_produtos);
        recycler_produtos.layoutManager = LinearLayoutManager(this)
        recycler_produtos.adapter = AdapterProduto(AddProdutos(),this)

    }

}